app/src/main/java/ch/epfl/polychef/Search.java

```diff
@@ -14,14 +14,43 @@
 import java.util.List;
 import java.util.function.BiFunction;
 
+/**
+ * An abstract class that represents the search in some Miniatures.
+ * @param <S> a {@link Miniatures} subclass
+ */
 public abstract class Search<S extends Miniatures> {
 
+    /**
+     * Gets the tag of the Miniature.
+     * @return the tag of the Miniature
+     */
     protected abstract String getTag();
 
+    /**
+     * Gets the miniature from the given data snapshot.
+     * @param dataSnapshot the data snapshot
+     * @return a miniature
+     */
     protected abstract S getValue(DataSnapshot dataSnapshot);
 
+    /**
+     * Gets the database name where we search.
+     * @return the database name where we search
+     */
     protected abstract String getDbName();
 
+    /**
+     * Gets the database where we search.
+     * @return the database where we search
+     */
+    protected abstract FirebaseDatabase getDatabase();
+
+    /**
+     * The main search method that sets listener for data changes.
```

Comment:
Not sure we need to specify about the listener since it is for a single event, but ok

Reply:
removed